src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,169 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidUserDataException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+
+    private final StorageDao storageDao = new StorageDaoImpl();
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+
+    @Test
+    public void register_existingLogin_notOk() {
+        User user = new User();
+        user.setLogin("TryMyCustomLogin");
+        user.setPassword("1234567");
+        user.setAge(20);
+
+        storageDao.add(user);
+
+        User newUser = new User();
+        newUser.setLogin("TryMyCustomLogin");
+        newUser.setPassword("12345678");
+        newUser.setAge(21);
+
+        assertThrows(InvalidUserDataException.class, () -> {
+            registrationService.register(newUser);
+        });
+
+    }
+
+    @Test
+    public void login_length_notOk() {
+        User user = new User();
+        user.setLogin("12345");
+        user.setPassword("12345678");
+        user.setAge(82);
+
+        assertThrows(InvalidUserDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    public void password_length_noOk() {
+        User user = new User();
+        user.setLogin("SuperLongLogin");
+        user.setPassword("short");
+        user.setAge(20);
+
+        assertThrows(InvalidUserDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    public void user_age_young_noOk() {
+        User user = new User();
+        user.setLogin("SuperLogin");
+        user.setPassword("LongPassword");
+        user.setAge(17);
+
+        assertThrows(InvalidUserDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    public void user_age_and_login_notOk() {
+        User user = new User();
+        user.setLogin("short");
+        user.setPassword("LongPassword");
+        user.setAge(3);
+
+        assertThrows(InvalidUserDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    public void user_age_and_password_notOk() {
+        User user = new User();
+        user.setLogin("LongLogin");
+        user.setPassword("abc");
+        user.setAge(5);
+
+        assertThrows(InvalidUserDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    public void user_login_noNull_noOk() {
```

Comment:
The test method name `user_login_noNull_noOk` contains a typo. It should be `user_login_null_notOk` for consistency and clarity.